src/priority_fee.rs

```diff
@@ -313,13 +410,75 @@ impl PriorityFeeTracker {
             very_high: max_percentile(&mut account_fees, &mut transaction_fees, 95),
             unsafe_max: max_percentile(&mut account_fees, &mut transaction_fees, 100),
         };
-        let end = std::time::Instant::now();
-        statsd_gauge!(
-            "get_priority_fee_estimates_time",
-            end.duration_since(start).as_nanos() as u64
-        );
         micro_lamport_priority_fee_estimates
     }
+
+    /*
+        Algo2: given the list of accounts the algorithm will:
+         1. collect all the transactions fees over n slots
+         2. for each specified account collect the fees and calculate the percentiles
+         4. choose maximum values for each percentile between all transactions and each account
+         */
+    fn calculation2(&self, accounts: &Vec<Pubkey>, include_vote: bool, lookback_period: &Option<u32>) -> MicroLamportPriorityFeeEstimates {
+        let lookback = calculate_lookback_size(&lookback_period, self.slot_cache.len());
+
+        let mut slots_vec = Vec::with_capacity(lookback);
+        self.slot_cache.copy_slots(&mut slots_vec);
+        slots_vec.sort();
+        slots_vec.reverse();
+
+        let mut fees = vec![];
+        let mut micro_lamport_priority_fee_estimates = MicroLamportPriorityFeeEstimates::default();
+
+        for slot in &slots_vec[..lookback] {
+            if let Some(slot_priority_fees) = self.priority_fees.get(slot)
+            {
+                if include_vote {
+                    fees.extend_from_slice(&slot_priority_fees.fees.vote_fees);
+                }
+                fees.extend_from_slice(&slot_priority_fees.fees.non_vote_fees);
+            }
+        }
+        estimate_max_values(&mut fees, &mut micro_lamport_priority_fee_estimates);
+
+        for account in accounts {
+            fees.clear();
+
+            for slot in &slots_vec[..lookback] {
+
+                if let Some(slot_priority_fees) = self.priority_fees.get(slot)
+                {
+                    let account_priority_fees = slot_priority_fees.account_fees.get(account);
+                    if let Some(account_priority_fees) = account_priority_fees {
+                        if include_vote {
+                            fees.extend_from_slice(&account_priority_fees.vote_fees);
+                        }
+                        fees.extend_from_slice(&account_priority_fees.non_vote_fees);
+                    }
+                }
+            }
+            estimate_max_values(&mut fees, &mut micro_lamport_priority_fee_estimates);
+        }
+        micro_lamport_priority_fee_estimates
+    }
+}
+
+fn estimate_max_values(
```

Comment:
this is relatively small/low impact, but in general as far as code goes I'm not a huge fan of impure functions for readability